mocks/sender.go

```diff
@@ -0,0 +1,88 @@
+package mocks
+
+import (
+	"context"
+	"sync"
+
+	pb "github.com/topfreegames/protos/eventsgateway/grpc/generated"
+)
+
+type MockSender struct {
+	totalSent           int
+	totalCalls          int
+	failureIndexesOrder [][]int64
+	firstCallEvents     []*pb.Event
+	eventsIdsFreqs      map[string]int
+	mutex               *sync.Mutex
+}
+
+func NewMockSender() *MockSender {
+	return &MockSender{
+		failureIndexesOrder: [][]int64{},
+		eventsIdsFreqs:      map[string]int{},
+		mutex:               &sync.Mutex{},
+	}
+}
+
+func (ms *MockSender) SendEvents(
+	ctx context.Context,
+	events []*pb.Event,
+) []int64 {
+	ms.mutex.Lock()
+	defer ms.mutex.Unlock()
+	defer func() { ms.totalCalls++ }()
```

Comment:
defers happen in reverse order, so the mutex is not protecting this change. I can't tell whether it's intentional or not, but I'd rather have `ms.totalCalls++` happen outside of a defer.